src/gluonnlp/metric/masked_accuracy.py

```diff
@@ -0,0 +1,304 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Masked accuracy metric."""
+
+import mxnet as mx
+from mxnet import ndarray
+
+__all__ = ['EvalMetric', 'MaskedAccuracy']
+
+def check_label_shapes(labels, preds, wrap=False, shape=False):
+    """Helper function for checking shape of label and prediction
+
+    Parameters
+    ----------
+    labels : list of `NDArray`
+        The labels of the data.
+
+    preds : list of `NDArray`
+        Predicted values.
+
+    wrap : boolean
+        If True, wrap labels/preds in a list if they are single NDArray
+
+    shape : boolean
+        If True, check the shape of labels and preds;
+        Otherwise only check their length.
+    """
+    if not shape:
+        label_shape, pred_shape = len(labels), len(preds)
+    else:
+        label_shape, pred_shape = labels.shape, preds.shape
+
+    if label_shape != pred_shape:
+        raise ValueError("Shape of labels {} does not match shape of "
+                         "predictions {}".format(label_shape, pred_shape))
+
+    if wrap:
+        if isinstance(labels, ndarray.ndarray.NDArray):
+            labels = [labels]
+        if isinstance(preds, ndarray.ndarray.NDArray):
+            preds = [preds]
+
+    return labels, preds
+
+class EvalMetric(object):
```

Comment:
Why do we need to create `EvalMetric ` instead of using `mxnet.metric.EvalMetric`? They are the same based on the code, right?

Reply:
Same comment in https://github.com/dmlc/gluon-nlp/pull/503#discussion_r247303242